File: android/src/main/java/org/maplibre/reactnative/components/layers/types/background/MLRNBackgroundLayerManager.java
package org.maplibre.reactnative.components.layers.types.background;

import com.facebook.react.uimanager.ThemedReactContext;

import org.maplibre.reactnative.components.layers.MLRNLayerManager;

public class MLRNBackgroundLayerManager extends MLRNLayerManager<MLRNBackgroundLayer> {
    public static final String REACT_CLASS = "MLRNBackgroundLayer";

    @Override
    public String getName() {
        return REACT_CLASS;
    }

    @Override
    protected MLRNBackgroundLayer createLayerViewInstance(ThemedReactContext reactContext) {
        return new MLRNBackgroundLayer(reactContext);
    }
}

File: android/src/main/java/org/maplibre/reactnative/components/layers/types/background/MLRNBackgroundLayerManager.kt
package org.maplibre.reactnative.components.layers.types.background

import com.facebook.react.uimanager.ThemedReactContext
import org.maplibre.reactnative.components.layers.MLRNLayerManager

class MLRNBackgroundLayerManager : MLRNLayerManager<MLRNBackgroundLayer?>() {
    override fun getName(): String {
        return REACT_CLASS
    }

    override fun createLayerViewInstance(reactContext: ThemedReactContext?): MLRNBackgroundLayer {
        return MLRNBackgroundLayer(reactContext)
    }

    companion object {
        const val REACT_CLASS: String = "MLRNBackgroundLayer"
    }
}
